themes/bootstrap3/templates/Recommend/SideFacets.phtml

```diff
@@ -28,7 +28,7 @@
 ?>
 <button class="close-offcanvas btn btn-link" data-toggle="offcanvas"><?=$this->transEsc('navigate_back') ?></button>
 <?php if ($results->getResultTotal() > 0): ?>
-  <h4><?=$this->transEsc(isset($this->overrideSideFacetCaption) ? $this->overrideSideFacetCaption : 'Narrow Search')?></h4>
+  <h4><?=$this->transEsc($this->slot('side-facet-caption')->set('Narrow Search')) ?></h4>
```

Comment:
This makes me wonder if it might make sense to support one more bit of alternate syntax, where `__invoke` accepts an optional second `$value` parameter, and if that is non-null, it returns the output of `set($value)`. Perhaps that makes the view helper code unnecessarily complicated/confusing, but on the flip side, I think it would make situations like this where a slot contains a brief default value a bit more readable in the templates.

Reply:
This could be easily done as long as it's not confusing that `$this->slot('foo', 'bar')` sets but `$this->slot('foo')` doesn't get. I don't believe we can detect no method after invoke.

Reply:
Actually, I think you might be able to make this work by implementing `__toString()` as an alias for `get()`. Then if you just retrieve the object and echo it without calling a subsequent mechanism, PHP will automatically call __toString and achieve the desired effect.